project1/thread_pool/task.h

```diff
@@ -0,0 +1,60 @@
+#ifndef PROJECT1_TASK_H
+#define PROJECT1_TASK_H
+
+#include <cstdint>
+
+namespace thread_pool {
+
+/**
+ * @brief Represents a task that can be run in a tread pool.
+ */
+class Task {
+ public:
+  /// Represents an opaque identifier for a task.
+  using Handle = uint32_t;
+
+  /**
+   * @brief Represents the status of a task.
+   */
+  enum class Status {
+    /// Task is running normally.
+    RUNNING,
+    /// Task has completed normally.
+    DONE,
+    /// Task has failed.
+    FAILED,
+  };
+
+  /**
+   * @brief Performs any necessary one-time setup when the task starts.
```

Comment:
What if certain tasks require different one-time setups? I'm slightly confused because different tasks could have different set up requirements? Maybe I'm thinking about this the wrong way.

Reply:
A specific one-time setup can be implemented for every `Task` subclass.